packages/core/src/modules/proofs/protocol/v2/__tests__/v2-indy-oob-proof-proposal.e2e.test.ts

```diff
@@ -0,0 +1,418 @@
+      state: ProofState.Done,
+    })
+
+    const proposalMessage = await aliceAgent.proofs.findProposalMessage(aliceProofExchangeRecord.id)
+    const requestMessage = await aliceAgent.proofs.findRequestMessage(aliceProofExchangeRecord.id)
+    const presentationMessage = await aliceAgent.proofs.findPresentationMessage(aliceProofExchangeRecord.id)
+
+    expect(proposalMessage).toBeInstanceOf(V2ProposePresentationMessage)
+    expect(requestMessage).toBeInstanceOf(V2RequestPresentationMessage)
+    expect(presentationMessage).toBeInstanceOf(V2PresentationMessage)
+  })
+
+  test('Alice start with oob proof proposal for Faber with aut-accept enabled', async () => {
+    const {
+      issuerAgent: faberAgent,
+      issuerReplay: faberReplay,
+      holderAgent: aliceAgent,
+      holderReplay: aliceReplay,
+      credentialDefinitionId,
+      issuerHolderConnectionId: faberConnectionId,
+    } = await setupAnonCredsTests({
+      issuerName: 'Faber oob Proofs proposal v2 - Auto Accept',
+      holderName: 'Alice oob Proofs proposal v2 - Auto Accept',
+      autoAcceptProofs: AutoAcceptProof.Always,
+      attributeNames: ['name', 'age'],
+    })
+
+    await issueLegacyAnonCredsCredential({
+      issuerAgent: faberAgent,
+      issuerReplay: faberReplay,
+      holderAgent: aliceAgent,
+      holderReplay: aliceReplay,
+      issuerHolderConnectionId: faberConnectionId,
+      offer: {
+        credentialDefinitionId,
+        attributes: [
+          {
+            name: 'name',
+            value: 'Alice',
+          },
+          {
+            name: 'age',
+            value: '99',
+          },
+        ],
+      },
+    })
+
+    agents = [aliceAgent, faberAgent]
+    testLogger.test('Alice creates oob proof proposal for faber')
+
+    const { message } = await aliceAgent.proofs.createProofProposal({
+      protocolVersion: 'v2',
+      proofFormats: {
+        indy: {
+          name: 'abc',
+          version: '1.0',
+          attributes: [
+            {
+              name: 'name',
+              value: 'Alice',
+              credentialDefinitionId,
+            },
+          ],
+          predicates: [
+            {
+              name: 'age',
+              predicate: '>=',
+              threshold: 50,
+              credentialDefinitionId,
+            },
+          ],
+        },
+      },
+      autoAcceptProof: AutoAcceptProof.ContentApproved,
+    })
+    const { outOfBandInvitation } = await aliceAgent.oob.createInvitation({
+      messages: [message],
+      autoAcceptConnection: true,
+    })
+
+    await faberAgent.oob.receiveInvitation(outOfBandInvitation)
+
+    await waitForProofExchangeRecordSubject(aliceReplay, {
+      state: ProofState.Done,
+    })
+    await waitForProofExchangeRecordSubject(faberReplay, {
+      state: ProofState.Done,
+    })
+  })
+
+  test('Alice start with oob proof proposal for Faber with auto-accept enabled and both agents having a mediator', async () => {
```

Comment:
Specific reason you added this test? We have a lot of tests that do almost the same and it's making updates harder. 

So if there's a specific reason for this test great, but otherwise we can maybe remove it?

Reply:
Right, For the oob proof proposal, we want to test
- After the proof exchange flow is completed, 
   - The holder should have all three proof messages `proposalMessage`, `requestMessage` , and `presentationMessage`
   - proof exchange record should have `nonNull` connection id.

I will remove the first and last tests and add the above assertion in 2nd test.